lib/public/UrlChecker.js

```diff
@@ -115,4 +122,33 @@ UrlChecker.prototype.resume = function()
 
 
 
+UrlChecker.prototype.clearCache = function()
+{
+	this.cache = {};
+};
+
+
+
+//::: PRIVATE FUNCTIONS
+
+
+
+function shouldQueryUrl(url, context)
+{
+	if (context.options.cacheResponses === true)
+	{
+		if (context.cache[url] === true)
+		{
+			return false;
```

Comment:
The PR is currently broken since this is half implemented. I am trying to figure out a way to test this via the black-box method you've been doing.